arrow-csv/src/reader/mod.rs

```diff
@@ -265,6 +266,14 @@ impl Format {
         self
     }
 
+    /// Whether to allow flexible lengths for records.
```

Comment:
```suggestion
    /// Whether to allow flexible number of columns for records.
    /// If enabled, records with not enough delimited values compared to expected
    /// number of columns will be padded with nulls for the missing columns.
```

Maybe something like this to make it clearer?

Reply:
Yes, I'm not sure how verbose to be in the naming. I've always called them ragged rows, but a colleague said that was confusing as well.

Maybe being even more explicit as `allow_trailing_empty_values_without_separarator` would be better?

Reply:
What do you think about `incomplete_rows`?

Reply:
I've pushed a new version with `allow_truncated_rows`. It's like incomplete rows, but maybe a bit clearer that it is only the end of the row that can be incomplete/missing.